### DISCUSSION TITLE
The Myth of BitClout Decentralization and Other Issues

### DISCUSSION THREAD

**Discussion by trevormil** [deso-protocol/dips#117] [Q&A]
Let's have a discussion about the road that BitClout is heading down.

Don't get it twisted. BitClout and its community is amazing, but the core team's actions have not been up to standard within the past months.

The Myth of Decentralization
- Verification: This latest $10 verification info pushed me over the edge. A truly decentralized platform would never ask for private info. Ever. This is not the crypto way. This is not decentralized. This verification process is leading BitClout down a path more towards centralization. Crypto is a magnificent technology. There are many better ways to implement verification than to give a central authority your data. I am building BitBadges which could potentially be used for decentralized verification. Even if BitBadges is not the solution, there are many better technology implementations than giving away your actual private info. 

- Developers - Developers and third party apps are getting no love. Hoping the "fuel on the fire" will change this, but besides the little drop down menu on bitclout.com with BitHunt and BitClout Pulse I believe, what actions have they done to integrate developers and third party apps?

- BitClout Identity - BitClout Identity is just a hack waiting to happen. When you sign into somewhere with access level 4 turned on, you are giving them the same power they would have if you just straight up gave them your private key. This is a disaster waiting to happen if they are promoting many third party apps to be built on top of BitClout. At least with mainstream crypto, the third party apps are companies with huge legal consequences. What is stopping the newest BitClout project created by Bob from Texas from funneling all your coins to them? Nothing. Plus, they are removing bitclout.com in the future. There will actually be no truly trusted frontend. Sure, some will gain trust over time, but my point is its way too easy to gain trust and too much power is in these third party app's hands. Maybe have two separate private keys? One that handles posts and likes and messages and one that handles funds? Almost all third party apps should never have access to funds. I don't know the current solution, but BitClout Identity is gonna end up terribly if it stays how it is.

- Infrastructure: They are not fully open source and decentralized yet. We have no access to see their database details and data they are collecting. Everything is routing through a centralized server they are running. Many more actions can occur to decentralize further. Right now, it is a good start and I hope they have future plans for further decentralization.

- Not a company? - According to their docs, they are "not a company, just code". How? VC funded, profits off creator coins (even people like Elon Musk they are making profits off of), and much more. Not consistent with what they claim to be. 

- Politics - There seems to be some clear favoritism with what celebrities they verify and what third party projects they choose to integrate. And they all are sparked by a big investment from one of the core team members, so they can profit off it. Certain projects get talked about. Certain projects get chosen to be featured in posts by the core team. Most are left in the dust. We need more equality.

Other Problems

Emoji Bots - Pretty self explanatory. These are getting out of hand.
Scams - Too many fake accounts and scammers on the platform.  Something needs to be done. And please fix this using crypto and not giving away private information.

I absolutely love the community on BitClout. I’ve made some really good friends there. But changes need to take place if I am to continue building upon it.

**Comment by tijno** [comment]
Valid concerns. 

What actions can be taken by the community to help address them?

Re specifics:

* Verification: This is not about verification i think - its about bots & overcoming onboarding friction allowing new users to start interacting with the platform without putting money in. It is optional and nodes dont have to follow bitclouts lead here, and i suspect many wont. Nodes can implement other mechanisms to solve the onboading vs bots challenge. The faucet project might offer such an opportunity. 

* Developers: Agree besides the two mentioned and a link to cloutvista there has been little integration. More can be done for sure. There have been a few shoutouts for projects. But i do think this is a two way exchange - eg what are developers contributing back to the open source code? The biggest projects build ontop of the bitclout eco system have contributed little to the opensource code base. Engagement with cips has reduced considerably, and community funding / sponsoring of open source development amounts to about $0.

* Identity: More granular permissions on Identity would indeed be amazing. Also the work thats been done by petrn on Derived keys is amazing. So i think improvements are being made. Also, i think Level 4 identity is not the same as giving someone your seeds. The user needs to be logged in to browser session for it to sign transactions. So server side or offline signing of transactions is not possible.

* Infrastructure: Yes its critical more progress is made here. Move away from admin switch profile & trusted block producers for me is key, but i suspect move to proof of stake is required for the latter before this will be possible. Re the data collected by Bitclout - most of that is visible in the code. Only thing not visible i think is the analytics. 

* Not a company: Not sure about this but in theory this is possible right ? Many projects are coded without being a company and my understanding is that there was no traditional vc funding - they bought bitclout like everyone else, but much earlier of course. 

* More equality: This would be great and ive at times got pretty frustrated with some getting profiled over others. I live the way clubhouse dealt with this where they had someone focussed on finding the amazing work done by creators and highlighting it on the weekly town hall. In relation to Bitclout, im not sure whether this is a right place right time issue or favouritism. We all are at most times aware of the projects and the people we engage with most, and may miss amazing stuff done by others that dont come across our own feeds. Discovery and curation us hard and could be an amazing place for a project. 

* Emoji bots: i thought they have stopped them a while ago. Ive not seen any recently. 

* Scams: yeah super annoying and wasteful. What crypto methods are there to prevent this?

Ok thats a long reply - keen to explore all of these and see what is possible.

**Reply by trevormil** [reply]
Yep, thanks for the reply and will try to address everything below. I know it is a lot of text.

Verification - Yea some nodes might go a different path. But most will follow the core BitClout team's decision. And the fact the core team took this decision, for me is unacceptable. No other mainstream crypto project would ask you for this kind of info. If they are not changing their mind on this, I'd like to see them or the community also implement an additional verification technology (BitBadges can offer a solution but many other solutions are out there) that will work with this feature side by side. This can't be the only thing they do with verification.

Developers - Yes, few have contributed, but also there are many that are just being ignored. I can speak firsthand. I messaged diamondhands multiple times. One response and then ghosted. CIP made a few weeks ago, nothing. I want to get BitBadges integrated, but this is a two way street and no one is showing support for my integration. Also, there have been many great projects in the past that have died off just because they were ignored - BitSwap, CloutPool, etc. All super interesting projects, but the core team doesn't give them the spotlight or capabilities to get integrated. Something needs to change here. They preach third party apps but don't do anything to promote them when good ones come along.

Identity - Yea, they do need to sign into the browser and your app for that, but like I said, what is stopping Bob from Texas from creating a new app, gaining trust, getting people to sign in, and funneling all funds to his wallet? Literally nothing. I bet you don't know 95% of the founders of any BitClout project right now in real life. How do you know their intentions? Money can change people and if just one of these third party apps sees a user that logs into their app w/ lots of funds, they can have a change of heart real quick. I'm not too sure on how it would work if they store all that info once they sign in once, but I believe that if they just store encryptedSeedHex and stuff, they don't need to be signed in. That gives them the power to access it forever. Could be mistaken though. Even if it is signed in, there is nothing stopping them from funneling all funds while signed in.

Infrastructure - Yea I am pretty satisfied with what they currently have, but definitely would like to see improvements in the near future. Or at least acknowledge them

Not a Company - I haven't done much research at all so definitely don't quote me on this. But I believe 40% of the $CLOUT supply was preminted and handed to the investors. I don't care who makes money. Even if they just bought it super early, that is the same as investing for me. But, you don't see Vitalik Buterin going and promoting a certain Ethereum NFT just so he can make money. I know it's a unique concept since BitClout inherently has monetization tied into it, but they seem way too centralized and there's way too much politics for a so called "decentralized" company.

Equality - Yep, discovery and curation is definitely needed w/ BitClout. My issue is mainly with the core team. Like I said above, you don't see Vitalik promoting NFTs. That only happens in for profit companies. I would be fine if it was just like a forum and discussion but they are literally making money in creator coin off promoting certain projects. diamondhands invests $50k into cloutpunk, promotes him, and makes triple his money. How is that not favoritism? That is a pump and dump scheme. 

Emoji Bots - Yea, just a long standing complaint I had. Definitely needs to be more features added though like approve users to send you a message or something. Haven't seen as many recently though. But then again, we have to ask the question. How is removing emoji bots from BitClout different than removing Trump from Twitter? Isn't that what BitClout stands for?

Scammers - Tough question but definitely solvable via crypto. BitBadges could solve this. If you trust someone to be a scammer spotter, they can issue scammer badges and you can ignore those users. If you suddenly don't trust the scammer spotter anymore, find another person you trust that is issuing badges.

So to recap: what actions can we take?

-Verification : Find a better solution and don't use the current private info one. Not the crypto way. BitBadges offers one. There's many. Anyone in the core team get in touch with me please and we can discuss this. For the community and nodes, adopt another project that gives better verification or don't give you private info to them.

-Developers: Nothing the community can really do about this. They are building great projects. Core team needs to give more spotlight to them. Not even just spotlight, more green lights for integration. More collaboration. My projects are written in JavaScript, React, etc. The official code base is written in Angular and Go (not so popular technologies). Developers like me aren't going to go through the hassle of learning two new languages just to create a PR. There has to be an easier way.

-Identity: For community, better education is needed. I hate to say it but someone needs to get hacked for us to see the true dangers of this. For the dev team, keep working on approving this. Maybe even disabling access level 4 or creating more access levels. Most apps don't need access to your funds. Don't give them this option.

-Infrastructure: Happy to help

- Equality and Not a Company: Nothing by the community can be done, but the core team needs to start acting more like a decentralized project. Right now, it seems more like a pump and dump scheme than a decentralized protocol with how the core team is handling themselves.

- Scammers: Need a way to see someone's reputation. I like what MemberCash is doing with the 5 star ratings. But for someone new to BitClout, they have no indication (possibly solved by a BitBadge) that someone has a history of scamming.


I know this was super long. We can definitely split these up into separate topics.

**Reply by trevormil** [reply]
For me, it's all about the principles. If you want to be a decentralized social media, the core team needs to act like decentralized founders.

**Reply by theparkerazzi** [reply]
Every single time there is a way for us to make any money they change all the rules. So now we can build on top of BitClout. Oh yay! I'm just waiting for the part where everyone gets screwed over. If it was up to me id take every creator to Privi or RUON, They seem to like the idea of putting us first. Bitclout can just be a revolving hackaton.

**Reply by theparkerazzi** [reply]
> For me, it's all about the principles. If you want to be a decentralized social media, the core team needs to act like decentralized founders.

I'm looking at building another platform, I'd love your help with it. We don't need to try to fix this. We can build a better one. They take and take and everyone just claps lol

**Comment by arhebbar** [comment]
I've said in one of my recent posts. The 10$ thing will definitely start or spur more scams and we need to be ready for it. On the other hand, doing this without that verification process would have been a recipe for disaster. But, I hope that 1) the private information shared by users is not stored on the blockchain (I understand a third party collected this) and 2) This is a temporary solution until a more decentralized solution comes thru (like the Clubhouse Referrals and some Voting mechanism).

Projects not getting enough support / encouragement and favoritism of some, that has been my biggest concern about the project. If this project has to scale, building a great developer ecosystem is the only way. Pouring fuel on fire of user onboarding is great, but if the projects don't get funded or encouraged, they'll all move on to other opportunities and only BitClout will be the loser on this. **How about 100$ for every commit that is approved?** That should get a lot of developers into the system. Each 100$ will definitely help you improve the ROI on the millions you are putting on user-onboarding for sure.

We are working on CreaTiers with some money we've bootstrapped and self-funded. We've got great feedback wherever we've shown, but even large whales within BitClout are hesitating to put the money that has been put on Cloutfeed and other projects even though they say it has that potential. But, not having a platform to vet and fund some projects is disappointing. 

**My solution:** The core team should run showcases and hackathons and contests to screen these projects and adopt some that are doing well. I don't want to mention names of projects and risk not mentioning other good ones. But, BitBadges looks like a missing piece of the puzzle worth going deep on. But, hearing this feedback from @trevormil  and that he's considering moving away is disappointing.

**Reply by trevormil** [reply]
Thanks. Love, love, love the hackathon/monetize every commit ideas. To clarify, not planning to move away from BitBadges, but definitely not investing any more money into the platform until they change some of their ways.

**Comment by FreeTrade** [comment]
Thanks for starting the discussion of this.

I do think we've seen a pivot away from, or maybe a deprioritization of decentralization. I'd guess that DH is aware that most users don't understand it well and doesn't attract new users in the way that Creator Coins, NFTs and $10 free does. I understand the need to attract a userbase, but it does leave me disappointed.

I don't think decentralization is being worked on right now - changing to POS and removing 'Trusted Block Producers remains unstarted on the 'todo' list. Removing or curtailing 'SWAPIDENTITY' is straightforward but hasn't seen any progress. Generally introducing identity login, then google login, email verification, and now id verification for $10 are each only small steps away from decentralization, but indicate the general direction.

My overriding position is that it's actually impossible to have proper decentralization with the creator coin model because the MEV will hugely disrupt creator coin transactions. A few very technical and insightful folks are aware of this, but for the most part it's too esoteric for most to really grasp. I'm interested to see if BitClout can offer any solution to this, but frankly I'm not optimistic they have one given DH's response when I raised the issue.

As for the point that 'the community' should be offering solutions to these problems . . . well I think DH raised $200m in BTC and the investors hold $1Bil in marketcap of CLOUT, so I think it's really up to them to solve the problems and fund the development, not a job to lay at the feet of community and volunteer developers. Raising awareness of an issue is a first step, so I think it is a service to raise awareness of issues whether or not you have a solution or intend to work on one. 

Fair point you make about Identity - this is solvable though using derived keys and delegated authority, as tijn says, progress is being made on this by petern, so that's one bright spot at least.

**Reply by tijno** [reply]
> As for the point that 'the community' should be offering solutions to these problems . . . well I think DH raised $200m in BTC and the investors hold $1Bil in marketcap of CLOUT, so I think it's really up to them to solve the problems

Its both. Without community development this thing will become centralised! 

In the end it was inevitable

> because the MEV will hugely disrupt creator coin transactions

MEV?

**Reply by FreeTrade** [reply]
>Its both. Without community development this thing will become centralised!

Kind of a catch 22 though - before it is decentralized, community devs will be reluctant to work on it. My view is the warchest should be used to get to full decentralization, at which point community devs might be more interested in contributing.

MEV - sorry, Miner Extractable Value. On a network like Ethereum, it's a problem, but tolerable if the value being extracted is .25% or .5% of a transaction. With a creator coin, where a single transaction can send a price up or down 50% or more, a miner re-ordering, or rejecting trxs can lead to miners extracting large parts of the value of the trx.

**Reply by tijno** [reply]
Agree 120% the 2m clout held from the initial 6m presale should be used for growth:

- growth of userbase
- growth of dev community
- deliver the vision of which decentralisation is part

And yes catch 22 indeed. 

MEV - thx!! You think this is actually happening right now ? That transactions are being reordered to benefit those that control the order ?

**Reply by FreeTrade** [reply]
>MEV - thx!! You think this is actually happening right now ? That transactions are being reordered to benefit those that control the order ?

I'd hope not! The transactions are ordered by 'Trusted Block Producers' right now - so we might trust them to order the transactions in the order in which they arrive. Once the block producers are not trusted, in a decentralized system for example, then that'll start happening right away.

**Reply by trevormil** [reply]
FreeTrade, thanks for the reply. I agree with all your points. Crypto is all about the principles, and even if these decisions seem small in scale right now, the fact they are making these issues shows their true stance. Decentralization will be a process, but to get to full decentralized, we can't be making decisions that take us more towards centralization.

Also if I were to commit to bitclout's main repos, I would have to learn two completely new languages (Go and Angular) that I have no familiarity with. It's the little things. I have no incentive to as a developer besides creator coin.

**Reply by kanshi** [reply]
> Agree 120% the 2m clout held from the initial 6m presale should be used for growth:
>     * growth of userbase
>     * growth of dev community
>     * deliver the vision of which decentralisation is part

Totally agreed. 
I suggested incentivizing community developers before user acquisition happens - perhaps in a decentralized manner to avoid already apparent scaling issues and the need to re-onboard those users once the bitclout.com fades out.

I'm also not sure why and how the $10 KYC should be used by the builders for their benefit, but I think this was just another miscommunication by DH. Triggered quite a few people tho.

**Reply by sull** [reply]
>  Generally introducing identity login, then google login, email verification, and now id verification for $10 are each only small steps away from decentralization, but indicate the general direction.

It could be related to Coinbase requirements and might signal that they plan to experiment with more than just a coin listing.

**Reply by arhebbar** [reply]
> Totally agreed.
> I suggested incentivizing community developers before user acquisition happens - perhaps in a decentralized manner to avoid already apparent scaling issues and the need to re-onboard those users once the bitclout.com fades out.

How about? $10, $50 and $100 per Small, Medium and Large Commit that is approved.

**Reply by trevormil** [reply]
Good idea. Another option could be a bounty system like GitCoin uses.

**Reply by AikonMishima** [reply]
> > Agree 120% the 2m clout held from the initial 6m presale should be used for growth:
> > * growth of userbase
> > * growth of dev community
> > * deliver the vision of which decentralisation is part
> 
> Totally agreed.
> I suggested incentivizing community developers before user acquisition happens - perhaps in a decentralized manner to avoid already apparent scaling issues and the need to re-onboard those users once the bitclout.com fades out.
> 
> I'm also not sure why and how the $10 KYC should be used by the builders for their benefit, but I think this was just another miscommunication by DH. Triggered quite a few people tho.

This. 

Running a node (with a BitClout.com clone or otherwise) can be lucrative in a number of ways, but the ramp up cost is quite large.

Research & Development (aka commits) and Ecosystem growth should be rewarded in an unbiased manner. This has been the foundation of other successful blockchain protocols / ecosystems.

**Comment by kanshi** [comment]
Thank you for putting this all together and raising the concerns in a form of a discussion. Excellent comments too :)

Helping community leaders get their own nodes, gave me some hints at the needs of the early / leading community node operators. I see them as key stakeholders of the "Decentralized Social Network" we are building.  

These needs seem to be very different from what the core devs are doing with the core node / repository / identity. 
The KYC / google images and auth being excellent examples, but there are also to the translations, moderation, layouts, plenty of additional features... and the fact that all want to build "on top" rather than "from scratch".

Here are some of my concerns re: decentralizing the social network:

1. None of the community nodes are able to "compete for traffic" with the main node, without full permissions from Identity. Users will simply not use the node with the Identity popups. They will naturally gravitate towards the central node. And its all about the frontend and what it does with it anyway.

Having deployed our own identity service at love4src we started identifying and working with these concerns. 
Basically, once we took away "I trust the core team with my seed because I have no choice and they gave it to me anyway lol" - the real concerns become visible and we as a community can work to address them. 
Notice, that these concerns should equally apply to the core node's identity and the frontend, as we have no visibility into how and who is running / having access to the core frontend along the layered stack that is being run. Would we know if these people change?

2. Core team seems to be incentivizing tools complementary to the core node, ignoring the need to deploy a decentralized node network to prepare for "the time bitclout.com dissapears". 
3. Trusted Block Producers mean that community nodes can't monetize / decentralize core network operations as there are no rewards from the transactions and I see no incentives in sight regarding the future PoS.

**Reply by trevormil** [reply]
Yep, totally agree on everything. Thanks for sharing your thoughts!

**Comment by trevormil** [comment]
I also believe that we need more education on BitClout Identity and the powers you are giving third party apps. BitClout's target audience is people who are not too technical. It's a social media platform. Besides the little pop up from Identity that explains permissions which is often ignored, no average user would understand the power being given.

This is in addition to needing to create more customized access levels.

**Comment by sull** [comment]
Regarding literal decentralization (as opposed to ideologies and realities of the situation and actions taken thus far) there should be a fund and/or incubator type of program to incentivize new nodes to launch. Obviously DH and key backers should be the primary source of funds to support network health and growth via more nodes running (as opposed to spending funds to attain more users).

Nodes can be behind the scenes (no frontend, no userbase, no additional purpose) and nodes can be like startups with business models. And anything in between. The spectrum of nodes should be outlined and all should have conditional support via such a fund/program.

**Reply by trevormil** [reply]
Agree. They are spending $10 for every new user that is onboarded whereas what really needs more funding is the dev community and making it more decentralized

**Reply by sull** [reply]
[nodes > users](https://love4src.com/posts/577c2752f47662197b677ccb6e4642ec360d08352bad678e3bb1b01899de1a40)

I think "gas on fire" focus on users will end up being a timing and priority mistake. I also think they know what they are doing and do not care about strengthening the decentralized aspect of BitClout nearly as much as  strengthening "BitClout"... call it... brand and platform (bitclout.com). Why?

Because they are under pressure to prevent $clout from crashing. It is propped up atm with a $100 floor. 
Eventually, the market needs to decide, not the buyback entity that orchestrates with the current exchanges. 

So, they are looking at this from a centralized perspective primarily.... hence more users and more $clout circulation. 

But these new users will all funnel to bitclout.com and since they officially plan to sunset their node once the network can be more independent of them, the challenge will be how to explain this to these new users who will equate it to like Jack Dorsey saying we are going to shut down twitter.com but you can go to critter.com or chirp.xyz etc. This is not a familiar concept to many and assuming these new users will be a mix of ppl, not just those acquainted with decentralization concepts (even if they are censorship-resistant  leaning a la parlor-esque gimmicks).

This is all to say that they are choosing to cross that bridge when they get there, rather than proactively do whats logical for the network in favor of not risking $clout collapse. 

When I say **nodes > users** I am also aware that tilting too much in either direction can have consequences. Still, more should be done asap to boot up a healthy node network at least in tandem with the new user campaign. Instead, nothing is being done, no focus at all on dev/node needs.

**Reply by arhebbar** [reply]
Technically, can they not do both? The Dev Community is fairly small. Other than verifying those who are doing development here and investing 1000-2000$ in their coins, what has the core team done for them. 

I would go even further and say even those communicating here need to be incentivized because we are doing is aimed to strengthen the dev community and the UX on the platform.
